### PR TITLE
Detect Zone info in node labels 

### PR DESCRIPTION
**Pull request by anathoodell**
# Description
Detect Zone info in node labels and return the list of nodes with this label-value pair.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1613|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility
- [ ] I have executed the relevant end-to-end test scenarios

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] k8s cluster add node labels to some workers, and dump the nodelist that the function returns

### CODE REVIEW

File: tests/shared/crclient/client.go [hunk 1]
```diff
@@ -122,6 +124,15 @@ func (f Client) listPodList(list *corev1.PodList) error {
 	return nil
 }
 
+func (f Client) listNodeList(list *corev1.NodeList) error {
+	for k, v := range f.Objects {
+		if k.Kind == "Node" {
+			list.Items = append(list.Items, *v.(*corev1.Node))
```
Comment (atye) [inline tests/shared/crclient/client.go:130]: Is there a chance for this type conversion to panic? Maybe something like `node, ok := *v.(*corev1.Node); ok` is safer.

**Comment by EvgenyUglov** [comment]
Looks like we need to use https://github.com/dell/csm/issues/1613 for our PRs as part of the feature.
Also, please update the summary to remove the branch name

File: tests/shared/crclient/client.go (hunk 1 above)
Comment (anathoodell) [reply tests/shared/crclient/client.go]: thanks Aaron. this is fixed in https://github.com/dell/csm-operator/pull/811/commits/21d5b069a5b340b78a52ae2ca24133dcab88f84f

**Review by EvgenyUglov — APPROVED** [review]

**Comment by anathoodell** [comment]
@EvgenyUglov  fixed the bugid and the title as suggested. thanks

**Pull request merged**